docs/building_blocks_for_flexible_content.md

```diff
@@ -0,0 +1,566 @@
+Blocks container applied so that everying in the column can be styled relative to one and other.
+
+## Indexing block content in search
+
+Only content in blocks that has a label of `content` will be indexed in site search.
+
+```yaml
+blocks:
+- type: govspeak
+  content: |
+    ## This will be a searchable header
+    here's some content that's searchable
+- type: button
+  text: Press me (not searchable)
+- type: header
+  content: searchable header
+```
+
+In the example above only the text in the govspeak and header blocks will be searchable.
+
+## Navigation Groups
+
+The top-level key `navigation_groups` contains an array of items with the keys `id` and `links`. The id is used to identify the group for navigation blocks like [main navigation](#main-navigation) and [side navigation](#side-navigation), which reference a navigation group with their `navigation_group_id` key. The links key is an array of `text`/`href` values, although a links item can itself contain a `links` key pointing to an array of links for a submenu (main navigation supports sub menus, side navigation does not)
+
+```yaml
+navigation_groups:
+- id: Top Menu
+  links:
+  - text: Goals
+    href: /landing-page/goals
+  - text: Tasks
+    href: /landing-page/tasks
+    links:
+      - text: Be kinder
+        href: "/landing-page/be-kinder"
+      - text: Exercise more
+        href: "/landing-page/exercise-more"
+      - text: Donate to charity
+        href: "/landing-page/donate-to-charity"
+      - text: Learn something new
+        href: "/landing-page/learn-something-new"
+      - text: Be thankful
+        href: "/landing-page/be-thankful"
+```
+
+## Extending a page (Whitehall Only)
+
+When editing a page in Whitehall, you may wish to use data from another page. Adding the top-level key `extends`, and the path of another landing page, will cause the data from the current block to be merged over the data from the extended block before publishing. This is a shallow merge (ie only top-level values are checked), and is currently used so that navigation groups can be specified in one page and used by sub-pages.
+
+### Example
+
+If you have a homepage (at /landing-page/homepage) like this:
+
+```yaml
+navigation_groups:
+- id: Top Menu
+  links:
+  - text: Homepage
+    href: /landing-page/homepage
+  - text: Child 1
+    href: /landing-page/child
+  - (etc)
+blocks:
+- type: main_navigation
+  navigation_group_id: Top Menu
+- type: govspeak
+  content: <p>Hello!</p>
+```
+
+and a child page (at /landing-page/child) like this:
+
+```yaml
+extends: /landing-page/homepage
+blocks:
+- type: main_navigation
+  navigation_group_id: Top Menu
+- type: govspeak
+  content: <p>Hello from the child!</p>
+```
+
+...the navigation_groups element will be copied into the child page when it is saved (allowing you to specify a shared menu in one page and simply republish all the child pages with a standard Whitehall rake task).
```

Comment:
TO DO: Should mention whether the `extends` attribute copies over everything from the parent or just `navigation_groups`.